runtime/lib/backend/common/Utils.hpp

```diff
@@ -262,16 +264,14 @@ constexpr auto has_gate(const SimulatorGateInfoDataT<size> &arr, const std::stri
     return false;
 }
 
-static inline auto simulateDraw(const std::vector<double> &probs, int8_t postselect) -> bool
+static inline auto simulateDraw(const std::vector<double> &probs, std::optional<int32_t> postselect)
+    -> bool
 {
-    // Check validity of postselect value
-    RT_FAIL_IF(std::abs(postselect) > 1, "Invalid postselect value");
+    if (postselect) {
+        RT_FAIL_IF(*postselect < 0 || *postselect > 1, "Invalid postselect value");
+        RT_FAIL_IF(probs[*postselect] == 0, "Probability of postselect value is 0");
 
-    // Return the postselect value, do not draw
-    if (postselect != -1) {
-        RT_FAIL_IF(probs[postselect] == 0, "Probability of postselect value is 0");
-
-        return postselect == 1 ? true : false;
+        return *postselect == 1 ? true : false;
```

Comment:
You may want to consider using `auto postselect_value = postselect.value_or(-1)` and get rid of these dereferences.

Reply:
But then again we are reviving the magic number (-1) ...

Reply:
Let's use:
```
    if (postselect) {
        auto postselect_value = postselect.value();

        RT_FAIL_IF(postselect_value < 0 || postselect_value > 1, "Invalid postselect value");
        RT_FAIL_IF(probs[postselect_value] == 0, "Probability of postselect value is 0");

        return postselect_value == 1 ? true : false;
    }
```